### PR TITLE
fix(components): [select] group error when custom option component

### LINKED ISSUE #16614: [Component] [option-group, select] Custom ElOption component breaks group
Bug Type: **`Component`**

## Environment

- Vue Version: `3.4.23`
- Element Plus Version: `2.7.1`
- Browser / OS: `Chrome 123.0.0.0 / macOS 13.5.1`
- Build Tool: `Vite`

### Related Component

- `el-option-group`
- `el-select`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLXNlbGVjdCB2LW1vZGVsPVwidmFsdWVcIiBwbGFjZWhvbGRlcj1cIlNlbGVjdFwiIHN0eWxlPVwid2lkdGg6IDI0MHB4XCI+XG4gICAgPGN1c3RvbS1vcHRpb25cbiAgICAgIGxhYmVsPVwiV2l0aG91dCBncm91cFwiXG4gICAgICB2YWx1ZT1cImZvb1wiXG4gICAgLz5cblxuICAgIDxlbC1vcHRpb24tZ3JvdXBcbiAgICAgIHYtZm9yPVwiZ3JvdXAgaW4gb3B0aW9uc1wiXG4gICAgICA6a2V5PVwiZ3JvdXAubGFiZWxcIlxuICAgICAgOmxhYmVsPVwiZ3JvdXAubGFiZWxcIlxuICAgID5cbiAgICAgIDxjdXN0b20tb3B0aW9uXG4gICAgICAgIHYtZm9yPVwiaXRlbSBpbiBncm91cC5vcHRpb25zXCJcbiAgICAgICAgOmtleT1cIml0ZW0udmFsdWVcIlxuICAgICAgICA6bGFiZWw9XCJpdGVtLmxhYmVsXCJcbiAgICAgICAgOnZhbHVlPVwiaXRlbS52YWx1ZVwiXG4gICAgICAvPlxuICAgIDwvZWwtb3B0aW9uLWdyb3VwPlxuICA8L2VsLXNlbGVjdD5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQgQ3VzdG9tT3B0aW9uIGZyb20gJy4vQ3VzdG9tT3B0aW9uLnZ1ZSc7XG5cbmNvbnN0IHZhbHVlID0gcmVmKCcnKVxuY29uc3Qgb3B0aW9ucyA9IFtcbiAge1xuICAgIGxhYmVsOiAnUG9wdWxhciBjaXRpZXMnLFxuICAgIG9wdGlvbnM6IFtcbiAgICAgIHtcbiAgICAgICAgdmFsdWU6ICdTaGFuZ2hhaScsXG4gICAgICAgIGxhYmVsOiAnU2hhbmdoYWknLFxuICAgICAgfSxcbiAgICAgIHtcbiAgICAgICAgdmFsdWU6ICdCZWlqaW5nJyxcbiAgICAgICAgbGFiZWw6ICdCZWlqaW5nJyxcbiAgICAgIH0sXG4gICAgXSxcbiAgfSxcbiAge1xuICAgIGxhYmVsOiAnQ2l0eSBuYW1lJyxcbiAgICBvcHRpb25zOiBbXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAnQ2hlbmdkdScsXG4gICAgICAgIGxhYmVsOiAnQ2hlbmdkdScsXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICB2YWx1ZTogJ1NoZW56aGVuJyxcbiAgICAgICAgbGFiZWw6ICdTaGVuemhlbicsXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICB2YWx1ZTogJ0d1YW5nemhvdScsXG4gICAgICAgIGxhYmVsOiAnR3Vhbmd6aG91JyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAnRGFsaWFuJyxcbiAgICAgICAgbGFiZWw6ICdEYWxpYW4nLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuXVxuPC9zY3JpcHQ+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJzcmMvQ3VzdG9tT3B0aW9uLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLW9wdGlvblxuICAgIDpsYWJlbD1cInByb3BzLmxhYmVsXCJcbiAgICA6dmFsdWU9XCJwcm9wcy52YWx1ZVwiXG4gID5cbiAgICB7e3Byb3BzLmxhYmVsfX0gLSBzb21lIGV4dHJhIHRleHRcbiAgPC9lbC1vcHRpb24+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IHNldHVwPlxuY29uc3QgcHJvcHMgPSBkZWZpbmVQcm9wcyh7XG4gIGxhYmVsOiB7XG4gICAgdHlwZTogU3RyaW5nLFxuICAgIGRlZmF1bHQ6ICcnLFxuICB9LFxuICB2YWx1ZSA6IHtcbiAgICB0eXBlOiBbU3RyaW5nLCBOdW1iZXJdLFxuICAgIGRlZmF1bHQ6IG51bGwsXG4gIH0sXG59KVxuPC9zY3JpcHQ+IiwiX28iOnt9fQ==)

### Steps to reproduce

Open the Select dropdown in the reproduction link and see, that groups are not displayed in the dropdown.

### What is Expected?

Using a custom wrapper component for ElOption should display the group, as it would with a plain ElOption component.

### What is actually happening?

Using a custom wrapper component for ElOption (to add further customization) inside an ElOptionGroup hides the group by mistake. 

### Additional comments

This was working before as expected. Broke with v2.5.0.

### PR DESCRIPTION
**Pull request by Liao-js**
fix https://github.com/element-plus/element-plus/issues/16614

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/select/src/option-group.vue [hunk 1]
```diff
@@ -57,22 +57,30 @@ export default defineComponent({
...
     )
 
+    const checkIsOption = (node) => {
+      return (
+        node.type &&
+        node.type.name === 'ElOption' &&
+        node.component &&
+        node.component.proxy
+      )
+    }
```
Comment (kooriookami) [inline packages/components/select/src/option-group.vue:67]:
```suggestion
    const isOption = (node) => node.type?.name === 'ElOption' && node.component?.proxy
```

File: packages/components/select/src/option-group.vue [hunk 2]
```diff
@@ -57,22 +57,30 @@ export default defineComponent({
...
       const children = []
       if (isArray(node.children)) {
         node.children.forEach((child) => {
-          if (
-            child.type &&
-            child.type.name === 'ElOption' &&
-            child.component &&
-            child.component.proxy
-          ) {
+          if (checkIsOption(child)) {
```
Comment (kooriookami) [inline packages/components/select/src/option-group.vue:74]:
```suggestion
          if (isOption(child)) {
```

File: packages/components/select/src/option-group.vue [hunk 3]
```diff
@@ -57,22 +57,30 @@ export default defineComponent({
...
             children.push(child.component.proxy)
           } else if (child.children?.length) {
             children.push(...flattedChildren(child))
           } else if (child.component?.subTree) {
-            children.push(...flattedChildren(child.component.subTree))
+            if (checkIsOption(child.component.subTree)) {
+              children.push(child.component.subTree.component.proxy)
+            } else {
+              children.push(...flattedChildren(child.component.subTree))
+            }
```
Comment (kooriookami) [inline packages/components/select/src/option-group.vue:83]: If there are multiple levels of nesting, it's better to use recursion.

**Review by kooriookami — CHANGES_REQUESTED** [review]

File: packages/components/select/src/option-group.vue (hunk 3 above)
Comment (Liao-js) [reply packages/components/select/src/option-group.vue:83]: OK

**Comment by kooriookami** [comment]
@Liao-js Look at this [Demo](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLXNlbGVjdCB2LW1vZGVsPVwidmFsdWVcIiBwbGFjZWhvbGRlcj1cIlNlbGVjdFwiIHN0eWxlPVwid2lkdGg6IDI0MHB4XCI+XG4gICAgPGN1c3RvbS1vcHRpb25cbiAgICAgIGxhYmVsPVwiV2l0aG91dCBncm91cFwiXG4gICAgICB2YWx1ZT1cImZvb1wiXG4gICAgLz5cblxuICAgIDxlbC1vcHRpb24tZ3JvdXBcbiAgICAgIHYtZm9yPVwiZ3JvdXAgaW4gb3B0aW9uc1wiXG4gICAgICA6a2V5PVwiZ3JvdXAubGFiZWxcIlxuICAgICAgOmxhYmVsPVwiZ3JvdXAubGFiZWxcIlxuICAgID5cbiAgICAgIDxjdXN0b20tb3B0aW9uXG4gICAgICAgIHYtZm9yPVwiaXRlbSBpbiBncm91cC5vcHRpb25zXCJcbiAgICAgICAgOmtleT1cIml0ZW0udmFsdWVcIlxuICAgICAgICA6bGFiZWw9XCJpdGVtLmxhYmVsXCJcbiAgICAgICAgOnZhbHVlPVwiaXRlbS52YWx1ZVwiXG4gICAgICAvPlxuICAgIDwvZWwtb3B0aW9uLWdyb3VwPlxuICA8L2VsLXNlbGVjdD5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQgQ3VzdG9tT3B0aW9uIGZyb20gJy4vQ3VzdG9tT3B0aW9uLnZ1ZSc7XG5cbmNvbnN0IHZhbHVlID0gcmVmKCcnKVxuY29uc3Qgb3B0aW9ucyA9IFtcbiAge1xuICAgIGxhYmVsOiAnUG9wdWxhciBjaXRpZXMnLFxuICAgIG9wdGlvbnM6IFtcbiAgICAgIHtcbiAgICAgICAgdmFsdWU6ICdTaGFuZ2hhaScsXG4gICAgICAgIGxhYmVsOiAnU2hhbmdoYWknLFxuICAgICAgfSxcbiAgICAgIHtcbiAgICAgICAgdmFsdWU6ICdCZWlqaW5nJyxcbiAgICAgICAgbGFiZWw6ICdCZWlqaW5nJyxcbiAgICAgIH0sXG4gICAgXSxcbiAgfSxcbiAge1xuICAgIGxhYmVsOiAnQ2l0eSBuYW1lJyxcbiAgICBvcHRpb25zOiBbXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAnQ2hlbmdkdScsXG4gICAgICAgIGxhYmVsOiAnQ2hlbmdkdScsXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICB2YWx1ZTogJ1NoZW56aGVuJyxcbiAgICAgICAgbGFiZWw6ICdTaGVuemhlbicsXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICB2YWx1ZTogJ0d1YW5nemhvdScsXG4gICAgICAgIGxhYmVsOiAnR3Vhbmd6aG91JyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAnRGFsaWFuJyxcbiAgICAgICAgbGFiZWw6ICdEYWxpYW4nLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuXVxuPC9zY3JpcHQ+XG4iLCJzcmMvUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTE2NjIxLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIlxuICB9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsInNyYy9lbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgY29uc3Qgc3R5bGVzID0gWydodHRwczovL3ByZXZpZXctMTY2MjEtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnLCAnaHR0cHM6Ly9wcmV2aWV3LTE2NjIxLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJ10ubWFwKChzdHlsZSkgPT4ge1xuICAgIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgICAgbGluay5ocmVmID0gc3R5bGVcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgICB9KVxuICB9KVxuICByZXR1cm4gUHJvbWlzZS5hbGwoc3R5bGVzKVxufSIsInNyYy9DdXN0b21PcHRpb24udnVlIjoiPHRlbXBsYXRlPlxyXG4gIDx0ZW1wbGF0ZT5cclxuICA8ZWwtb3B0aW9uXHJcbiAgICA6bGFiZWw9XCJwcm9wcy5sYWJlbFwiXHJcbiAgICA6dmFsdWU9XCJwcm9wcy52YWx1ZVwiXHJcbiAgPlxyXG4gICAge3twcm9wcy5sYWJlbH19IC0gc29tZSBleHRyYSB0ZXh0XHJcbiAgPC9lbC1vcHRpb24+XHJcbiAgPC90ZW1wbGF0ZT5cclxuPC90ZW1wbGF0ZT5cclxuXHJcbjxzY3JpcHQgc2V0dXA+XHJcbmNvbnN0IHByb3BzID0gZGVmaW5lUHJvcHMoe1xyXG4gIGxhYmVsOiB7XHJcbiAgICB0eXBlOiBTdHJpbmcsXHJcbiAgICBkZWZhdWx0OiAnJyxcclxuICB9LFxyXG4gIHZhbHVlIDoge1xyXG4gICAgdHlwZTogW1N0cmluZywgTnVtYmVyXSxcclxuICAgIGRlZmF1bHQ6IG51bGwsXHJcbiAgfSxcclxufSlcclxuPC9zY3JpcHQ+IiwiX28iOnsic2hvd0hpZGRlbiI6dHJ1ZSwic3R5bGVTb3VyY2UiOiJodHRwczovL3ByZXZpZXctMTY2MjEtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MifX0=)~ When there're nested templates. It won't work.

**Comment by Liao-js** [comment]
> @Liao-js Look at this [Demo](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGVsLXNlbGVjdCB2LW1vZGVsPVwidmFsdWVcIiBwbGFjZWhvbGRlcj1cIlNlbGVjdFwiIHN0eWxlPVwid2lkdGg6IDI0MHB4XCI+XG4gICAgPGN1c3RvbS1vcHRpb25cbiAgICAgIGxhYmVsPVwiV2l0aG91dCBncm91cFwiXG4gICAgICB2YWx1ZT1cImZvb1wiXG4gICAgLz5cblxuICAgIDxlbC1vcHRpb24tZ3JvdXBcbiAgICAgIHYtZm9yPVwiZ3JvdXAgaW4gb3B0aW9uc1wiXG4gICAgICA6a2V5PVwiZ3JvdXAubGFiZWxcIlxuICAgICAgOmxhYmVsPVwiZ3JvdXAubGFiZWxcIlxuICAgID5cbiAgICAgIDxjdXN0b20tb3B0aW9uXG4gICAgICAgIHYtZm9yPVwiaXRlbSBpbiBncm91cC5vcHRpb25zXCJcbiAgICAgICAgOmtleT1cIml0ZW0udmFsdWVcIlxuICAgICAgICA6bGFiZWw9XCJpdGVtLmxhYmVsXCJcbiAgICAgICAgOnZhbHVlPVwiaXRlbS52YWx1ZVwiXG4gICAgICAvPlxuICAgIDwvZWwtb3B0aW9uLWdyb3VwPlxuICA8L2VsLXNlbGVjdD5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5pbXBvcnQgQ3VzdG9tT3B0aW9uIGZyb20gJy4vQ3VzdG9tT3B0aW9uLnZ1ZSc7XG5cbmNvbnN0IHZhbHVlID0gcmVmKCcnKVxuY29uc3Qgb3B0aW9ucyA9IFtcbiAge1xuICAgIGxhYmVsOiAnUG9wdWxhciBjaXRpZXMnLFxuICAgIG9wdGlvbnM6IFtcbiAgICAgIHtcbiAgICAgICAgdmFsdWU6ICdTaGFuZ2hhaScsXG4gICAgICAgIGxhYmVsOiAnU2hhbmdoYWknLFxuICAgICAgfSxcbiAgICAgIHtcbiAgICAgICAgdmFsdWU6ICdCZWlqaW5nJyxcbiAgICAgICAgbGFiZWw6ICdCZWlqaW5nJyxcbiAgICAgIH0sXG4gICAgXSxcbiAgfSxcbiAge1xuICAgIGxhYmVsOiAnQ2l0eSBuYW1lJyxcbiAgICBvcHRpb25zOiBbXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAnQ2hlbmdkdScsXG4gICAgICAgIGxhYmVsOiAnQ2hlbmdkdScsXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICB2YWx1ZTogJ1NoZW56aGVuJyxcbiAgICAgICAgbGFiZWw6ICdTaGVuemhlbicsXG4gICAgICB9LFxuICAgICAge1xuICAgICAgICB2YWx1ZTogJ0d1YW5nemhvdScsXG4gICAgICAgIGxhYmVsOiAnR3Vhbmd6aG91JyxcbiAgICAgIH0sXG4gICAgICB7XG4gICAgICAgIHZhbHVlOiAnRGFsaWFuJyxcbiAgICAgICAgbGFiZWw6ICdEYWxpYW4nLFxuICAgICAgfSxcbiAgICBdLFxuICB9LFxuXVxuPC9zY3JpcHQ+XG4iLCJzcmMvUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTE2NjIxLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIlxuICB9XG59IiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsInNyYy9lbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgY29uc3Qgc3R5bGVzID0gWydodHRwczovL3ByZXZpZXctMTY2MjEtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnLCAnaHR0cHM6Ly9wcmV2aWV3LTE2NjIxLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJ10ubWFwKChzdHlsZSkgPT4ge1xuICAgIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgICAgbGluay5ocmVmID0gc3R5bGVcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgICB9KVxuICB9KVxuICByZXR1cm4gUHJvbWlzZS5hbGwoc3R5bGVzKVxufSIsInNyYy9DdXN0b21PcHRpb24udnVlIjoiPHRlbXBsYXRlPlxyXG4gIDx0ZW1wbGF0ZT5cclxuICA8ZWwtb3B0aW9uXHJcbiAgICA6bGFiZWw9XCJwcm9wcy5sYWJlbFwiXHJcbiAgICA6dmFsdWU9XCJwcm9wcy52YWx1ZVwiXHJcbiAgPlxyXG4gICAge3twcm9wcy5sYWJlbH19IC0gc29tZSBleHRyYSB0ZXh0XHJcbiAgPC9lbC1vcHRpb24+XHJcbiAgPC90ZW1wbGF0ZT5cclxuPC90ZW1wbGF0ZT5cclxuXHJcbjxzY3JpcHQgc2V0dXA+XHJcbmNvbnN0IHByb3BzID0gZGVmaW5lUHJvcHMoe1xyXG4gIGxhYmVsOiB7XHJcbiAgICB0eXBlOiBTdHJpbmcsXHJcbiAgICBkZWZhdWx0OiAnJyxcclxuICB9LFxyXG4gIHZhbHVlIDoge1xyXG4gICAgdHlwZTogW1N0cmluZywgTnVtYmVyXSxcclxuICAgIGRlZmF1bHQ6IG51bGwsXHJcbiAgfSxcclxufSlcclxuPC9zY3JpcHQ+IiwiX28iOnsic2hvd0hpZGRlbiI6dHJ1ZSwic3R5bGVTb3VyY2UiOiJodHRwczovL3ByZXZpZXctMTY2MjEtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MifX0=)~ When there're nested templates. It won't work.

template is rendered into the tag template, so it cannot be seen
微信图片_20240424150045

**Review by kooriookami — APPROVED** [review]

File: packages/components/select/src/option-group.vue [hunk 4]
```diff
@@ -57,25 +57,24 @@ export default defineComponent({
       children.value.some((option) => option.visible === true)
     )
 
+    const isOption = (node) =>
+      node.type?.name === 'ElOption' && node.component?.proxy
```
Comment (btea) [inline packages/components/select/src/option-group.vue:61]: According to the name semantics, the function should return a boolean value, perhaps the following adjustment will make it clearer.
```suggestion
      node.type?.name === 'ElOption' && !!node.component?.proxy
```

File: packages/components/select/src/option-group.vue [hunk 5]
```diff
@@ -57,25 +57,24 @@ export default defineComponent({
...
+
     // get all instances of options
     const flattedChildren = (node) => {
+      const Nodes = isArray(node) ? node : [node]
```
Comment (btea) [inline packages/components/select/src/option-group.vue:65]: There is an `ensureArray` function in `@element-plus/utils`, which is more convenient to use directly.
```suggestion
      const Nodes = ensureArray(node)
```

**Review by btea — COMMENTED** [review]
Can you add a tese case?

**Comment by Liao-js** [comment]
> Can you add a tese case?

OK

**Review by btea — APPROVED** [review]

**Pull request merged**